internal/cmd/byok/command_register.go

```diff
@@ -0,0 +1,222 @@
+package byok
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"regexp"
+
+	"github.com/spf13/cobra"
+
+	byokv1 "github.com/confluentinc/ccloud-sdk-go-v2/byok/v1"
+	pcmd "github.com/confluentinc/cli/internal/pkg/cmd"
+	"github.com/confluentinc/cli/internal/pkg/errors"
+	"github.com/confluentinc/cli/internal/pkg/examples"
+	"github.com/confluentinc/cli/internal/pkg/output"
+	"github.com/confluentinc/cli/internal/pkg/resource"
+	"github.com/confluentinc/cli/internal/pkg/utils"
+
+	"github.com/aws/aws-sdk-go/aws/arn"
+)
+
+var encryptionKeyPolicyAWS = template.Must(template.New("encryptionKeyPolicyAWS").Parse(`{
+	"Sid" : "Allow Confluent accounts to use the key",
+	"Effect" : "Allow",
+	"Principal" : {
+		"AWS" : [{{range $i, $e := .}}{{if $i}},{{end}}
+			"{{$e}}"{{end}}
+		]
+	},
+	"Action" : [ "kms:Encrypt", "kms:Decrypt", "kms:ReEncrypt*", "kms:GenerateDataKey*", "kms:DescribeKey" ],
+	"Resource" : "*"
+}, {
+	"Sid" : "Allow Confluent accounts to attach persistent resources",
+	"Effect" : "Allow",
+	"Principal" : {
+		"AWS" : [{{range $i, $e := .}}{{if $i}},{{end}}
+			"{{$e}}"{{end}}
+		]
+	},
+	"Action" : [ "kms:CreateGrant", "kms:ListGrants", "kms:RevokeGrant" ],
+	"Resource" : "*"
+}`))
+
+const keyVaultCryptoServiceEncryptionUser = "e147488a-f6f5-4113-8e2d-b22465e65bf6"
+const keyVaultReader = "21090545-7ca7-4776-b22c-e363652d74d2"
+
+func (c *command) newRegisterCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "register <key>",
+		Short: "Register a self-managed key in Confluent Cloud.",
+		Args:  cobra.ExactArgs(1),
+		RunE:  c.register,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Register a new self-managed encryption key for AWS:",
+				Code: `confluent byok register "arn:aws:kms:us-west-2:111122223333:key/1234abcd-12ab-34cd-56ef-1234567890ab"`,
+			},
+			examples.Example{
+				Text: "Register a new self-managed encryption key for Azure:",
+				Code: `confluent byok register "https://a-vault.vault.azure.net/keys/a-key/00000000000000000000000000000000" --tenantid "00000000-0000-0000-0000-000000000000" --key-vault-id "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/a-resourcegroups/providers/Microsoft.KeyVault/vaults/a-vault"`,
+			},
+		),
+	}
+
+	cmd.Flags().String("key-vault-id", "", "The ID of the Azure Key Vault where the key is stored.")
+	cmd.Flags().String("tenant-id", "", "The ID of the Azure Active Directory tenant that the key vault belongs to.")
+	cmd.MarkFlagsRequiredTogether("key-vault-id", "tenant-id")
```

Comment:
Let's move this underneath the list of flag definitions.

Reply:
Fixed.